metaflow/runtime.py

```diff
@@ -238,8 +238,28 @@ def execute(self):
                                         'by the end of flow.')
 
     def _killall(self):
+        # If we are here, all children have received a signal and are shutting down.
+        # We want to give them an opportunity to do so and then kill
+        live_workers = []
+        temp_workers = {}
         for worker in self._workers.values():
-            worker.kill()
+            temp_workers[worker] = 1
+        live_workers = temp_workers.keys()
+        now = int(time.time())
+        self._logger('Terminating %d active tasks...' % len(live_workers),
+                     system_msg=True, bad=True)
+        while live_workers and int(time.time()) - now < 5:
+            # While not all workers are dead and we have waited less than 5 seconds
+            live_workers = [worker for worker in live_workers if not worker.clean()]
```

Comment:
Feel free to reject this suggestion - 

This seems like a good use-case for `filter`:
live_workers = filter(lambda worker: not worker.clean(), live_workers)
I don't know the mutability guarantees for list iteration, so I'm not sure if using a function like `filter` is better.

As bonus it might work on set(s) or list(s).

Reply:
I read a bit that filter may be a tad slower (not a huge deal here though) but I find list comprehension to be more readable so if OK, I would prefer to leave it as such.